hebiros_gazebo_plugin/plugin_hebiros/hebiros_gazebo_group.cpp

```diff
@@ -0,0 +1,237 @@
+#include "hebiros_gazebo_group.h"
+
+namespace hebi {
+namespace sim {
+namespace plugin {
+
+HebirosGazeboGroup::HebirosGazeboGroup(std::string name,
+  const std::vector<hebi::sim::Joint*>& joints_,
+  std::shared_ptr<ros::NodeHandle> n) {
+
+  int size = joints_.size();
+
+  feedback.name.resize(size);
+  for (auto joint : joints_)
+  {
+    joints.push_back(joint);
+    feedback.name.push_back(joint->getName());
+  }
+
+  feedback.position.resize(size);
+  feedback.motor_winding_temperature.resize(size);
+  feedback.motor_housing_temperature.resize(size);
+  feedback.board_temperature.resize(size);
+  feedback.velocity.resize(size);
+  feedback.effort.resize(size);
+  // Default, return "nan" for feedback, until we set something!
+  feedback.position_command.resize(size, std::numeric_limits<float>::quiet_NaN());
+  feedback.velocity_command.resize(size, std::numeric_limits<float>::quiet_NaN());
+  feedback.effort_command.resize(size, std::numeric_limits<float>::quiet_NaN());
+  feedback.accelerometer.resize(size);
+  feedback.gyro.resize(size);
+
+  feedback_pub = n->advertise<hebiros::FeedbackMsg>(
+    "hebiros_gazebo_plugin/feedback/" + name, 100);
+
+  this->name = name;
+
+  ros::Time current_time = ros::Time::now();
+  this->start_time = current_time;
+  this->prev_time = current_time;
+  this->prev_feedback_time = current_time;
+
+  this->command_sub = n->subscribe<hebiros::CommandMsg>("hebiros_gazebo_plugin/command/"+name, 100,
+    boost::bind(&HebirosGazeboGroup::SubCommand, this, _1));
+
+  this->acknowledge_srv =
+    n->advertiseService<std_srvs::Empty::Request, std_srvs::Empty::Response>(
+    "hebiros_gazebo_plugin/acknowledge/"+name, boost::bind(
+    &HebirosGazeboGroup::SrvAcknowledge, this, _1, _2));
+
+  this->command_lifetime_srv =
+    n->advertiseService<hebiros::SetCommandLifetimeSrv::Request, hebiros::SetCommandLifetimeSrv::Response>(
+    "hebiros_gazebo_plugin/set_command_lifetime/"+name, boost::bind(
+    &HebirosGazeboGroup::SrvSetCommandLifetime, this, _1, _2));
+
+  this->feedback_frequency_srv =
```

Comment:
I've begun liking the idea of having a very simple constructor (and delegating any "complicated" field initialization to a method _before_ the constructor is invoked - something like a `Type::createType(args)` which creates a shared_ptr) - that way you can have the ctor inlined. I guess it's mainly for making `std::make_shared<...>(...)` easier to use, but I forget which C++ version that is in (make_unique was added in a different C++ version than make_shared confusingly enough) and what version we require here, so this is more of a note for the future I suppose

Reply:
Also coincidentally enough, it looks like instances of this class are held as a `shared_ptr`...

Reply:
Me too -- I didn't really refactor most of the `hebiros_gazebo_group.cpp` code; this is primarily legacy here.